aten/src/ATen/native/cuda/LogcumsumexpKernel.cu

```diff
@@ -11,24 +11,106 @@
 
 namespace at::native {
 
+// custom min and max to be used in logcumsumexp for complex arguments
+template <typename scalar_t>
+__host__ __device__ c10::complex<scalar_t> _logcumsumexp_minmax(const c10::complex<scalar_t>& x, const c10::complex<scalar_t>& y, const bool min) {
```

Comment:
The min here could be a template argument (since it's only used for dispatching the function in the first place). Does doing that also cause compile time explosion?

Reply:
It adds a little bit of compile time, but not significant. The increase might very well from just the fluctuation in my computer.